include/psen_scan_v2/monitoring_frame_msg.h

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2020 Pilz GmbH & Co. KG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+#ifndef PSEN_SCAN_V2_MONITORING_FRAME_MSG_H
+#define PSEN_SCAN_V2_MONITORING_FRAME_MSG_H
+
+#include <cstdint>
+#include <functional>
+#include <map>
+#include <sstream>
+#include <vector>
+
+#include <gtest/gtest_prod.h>
+
+#include "psen_scan_v2/raw_processing.h"
+#include "psen_scan_v2/raw_scanner_data.h"
+
+namespace psen_scan_v2
+{
+static constexpr uint32_t OP_CODE_MONITORING_FRAME{ 0xCA };
+static constexpr uint32_t ONLINE_WORKING_MODE{ 0x00 };
+static constexpr uint32_t GUI_MONITORING_TRANSACTION{ 0x05 };
+static constexpr uint32_t MAX_SCANNER_ID{ 0x03 };
+
+class FieldHeader
+{
+public:
+  using Id = uint8_t;
+  using Length = uint16_t;
+
+public:
+  FieldHeader(std::istringstream& is);
```

Comment:
I very much dislike this construction. It also adds so much complexity that `length_` could end up being 65535. However this would clearly be suspicious.

Reply:
See above discussion